src/directives/draw.js

```diff
@@ -0,0 +1,99 @@
+(() => {
+  const directiveName = 'lfDraw';
+
+  angular.module('ui-leaflet')
+  .config(function($provide) {
+    /*
+    Extend root leaflet directive,
+    */
+    return $provide.decorator('leafletDirective', function($delegate) {
+      var directive;
+      directive = $delegate[0];
+      // directive.scope[directiveName] = '=?'; //the way it should be
+      //the above should derrive $$isolateBindings this without the below hack
+      //workaround from http://plnkr.co/edit/CRxhX6?p=preview
+      directive.$$isolateBindings[directiveName] = {
+        attrName: directiveName,
+        mode: '=',
+        optional: true
+      };
+
+      return $delegate;
+    });
+  })
+  .directive(directiveName, function(leafletLogger, leafletData, leafletHelpers, leafletIterators, leafletDrawEvents, $timeout, $q) {
+    var $log, errorHeader, isDefined;
+    $log = leafletLogger;
+    isDefined = leafletHelpers.isDefined;
+    errorHeader = leafletHelpers.errorHeader;
+    return {
+      restrict: 'A',
+      scope: false,
+      replace: false,
+      require: ['leaflet'],
+      controller: function($scope) {
+        this._deferredDrawTool = $q.defer();
+        this.getDrawTool = function() {
+          return this._deferredDrawTool.promise;
+        };
+        this.getScope = function() {
+          return $scope;
+        };
+      },
+      link: function(scope, element, attrs, controller) {
+        var _deferred, _featureGroup, _optionsEditedInDirective, leafletScope, mapController;
+        mapController = controller[0];
+        leafletScope = mapController.getLeafletScope();
+        _featureGroup = void 0;
+        _optionsEditedInDirective = false;
+        _deferred = void 0;
+        return leafletScope.$watchCollection(directiveName, function() {
+          var options;
+          if (!_deferred || _deferred.resolvedDefer) {
+            _deferred = $q.defer();
+          }
+          if (_optionsEditedInDirective) {
+            return;
+          }
+          options = leafletScope[directiveName] || {};
+          if (isDefined(options.control)) {
+            options.control.promised(_deferred.promise);
```

Comment:
To do this can go away once I make leafletData and directiveControls extendable.